### ISSUE TITLE
Failed to create employee record due to missing Content-Type

### ISSUE THREAD

**Issue by ducduyn31** [merge-api/merge-sdk-kotlin#5]
## Expected Behavior
When making create employee request, it should be successful

## Current Behavior
When making create employee request, it is failing due to missing Content-Type. When listing employees, it is still successful.

## Possible Solution
Providing Content-Type when making request in `ApiClient.kt`

## Steps to Reproduce
Executing this below piece of code
```
val employeesApi = EmployeesApi()
employeesApi.setApiKey(apiKey)
employeesApi.setAccountToken(accountToken)

employeesApi.employeesCreate(
    EmployeesApi.EmployeesCreateRequest(
        employeeEndpointRequest = EmployeeEndpointRequest(
            model = EmployeeRequest(
                firstName = "John",
                lastName = "Doe",
            )
        )
    )
)
```

## Context
- SDK Version: 2.0.1
- OS: macOS 12.6

## Detailed Stacktrace
```
Fail to prepare request body for sending. 
The body type is: class dev.merge.client.hris.models.EmployeeEndpointRequest, with Content-Type: null.

If you expect serialized body, please check that you have installed the corresponding plugin(like `ContentNegotiation`) and set `Content-Type` header.
java.lang.IllegalStateException: Fail to prepare request body for sending. 
The body type is: class dev.merge.client.hris.models.EmployeeEndpointRequest, with Content-Type: null.

If you expect serialized body, please check that you have installed the corresponding plugin(like `ContentNegotiation`) and set `Content-Type` header.
	at io.ktor.client.plugins.HttpSend$Plugin$install$1.invokeSuspend(HttpSend.kt:88)
	at io.ktor.client.plugins.HttpSend$Plugin$install$1.invoke(HttpSend.kt)
	at io.ktor.client.plugins.HttpSend$Plugin$install$1.invoke(HttpSend.kt)
	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
	at io.ktor.util.pipeline.SuspendFunctionGun.proceedWith(SuspendFunctionGun.kt:91)
	at io.ktor.client.plugins.HttpCallValidator$Companion$install$1.invokeSuspend(HttpCallValidator.kt:126)
	at io.ktor.client.plugins.HttpCallValidator$Companion$install$1.invoke(HttpCallValidator.kt)
	at io.ktor.client.plugins.HttpCallValidator$Companion$install$1.invoke(HttpCallValidator.kt)
	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
	at io.ktor.client.plugins.HttpRequestLifecycle$Plugin$install$1.invokeSuspend(HttpRequestLifecycle.kt:35)
	at io.ktor.client.plugins.HttpRequestLifecycle$Plugin$install$1.invoke(HttpRequestLifecycle.kt)
	at io.ktor.client.plugins.HttpRequestLifecycle$Plugin$install$1.invoke(HttpRequestLifecycle.kt)
	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
	at io.ktor.util.pipeline.SuspendFunctionGun.execute$ktor_utils(SuspendFunctionGun.kt:101)
	at io.ktor.util.pipeline.Pipeline.execute(Pipeline.kt:77)
	at io.ktor.client.HttpClient.execute$ktor_client_core(HttpClient.kt:187)
	at io.ktor.client.statement.HttpStatement.executeUnsafe(HttpStatement.kt:107)
	at io.ktor.client.statement.HttpStatement.execute(HttpStatement.kt:46)
	at io.ktor.client.statement.HttpStatement.execute(HttpStatement.kt:61)
	at dev.merge.client.shared.ApiClient.request(ApiClient.kt:219)
	at dev.merge.client.shared.ApiClient.jsonRequest(ApiClient.kt:138)
	at dev.merge.client.hris.apis.EmployeesApi.employeesCreate$suspendImpl(EmployeesApi.kt:442)
	at dev.merge.client.hris.apis.EmployeesApi.employeesCreate(EmployeesApi.kt)
	at BasicTest$testAllCategoriesSimple$1.invokeSuspend(BasicTest.kt:33)
	at BasicTest$testAllCategoriesSimple$1.invoke(BasicTest.kt)
	at BasicTest$testAllCategoriesSimple$1.invoke(BasicTest.kt)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTestCoroutine$2.invokeSuspend(TestBuilders.kt:212)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTestCoroutine$2.invoke(TestBuilders.kt)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTestCoroutine$2.invoke(TestBuilders.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:55)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:112)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTestCoroutine(TestBuilders.kt:211)
	at kotlinx.coroutines.test.TestBuildersKt.runTestCoroutine(Unknown Source)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$1$1.invokeSuspend(TestBuilders.kt:167)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$1$1.invoke(TestBuilders.kt)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$1$1.invoke(TestBuilders.kt)
	at kotlinx.coroutines.test.TestBuildersJvmKt$createTestResult$1.invokeSuspend(TestBuildersJvm.kt:13)
	(Coroutine boundary)
	at io.ktor.client.plugins.HttpCallValidator$Companion$install$1.invokeSuspend(HttpCallValidator.kt:126)
	at io.ktor.client.plugins.HttpRequestLifecycle$Plugin$install$1.invokeSuspend(HttpRequestLifecycle.kt:35)
	at io.ktor.client.statement.HttpStatement.executeUnsafe(HttpStatement.kt:107)
	at io.ktor.client.statement.HttpStatement.execute(HttpStatement.kt:46)
	at dev.merge.client.hris.apis.EmployeesApi.employeesCreate$suspendImpl(EmployeesApi.kt:442)
	at BasicTest$testAllCategoriesSimple$1.invokeSuspend(BasicTest.kt:33)
	at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTestCoroutine$2.invokeSuspend(TestBuilders.kt:212)

...
```

## Possible Implementation
in `ApiClient.kt:158:`
```
if (requestConfig.method in listOf(RequestMethod.PUT, RequestMethod.POST, RequestMethod.PATCH)) {
    this.contentType(ContentType.Application.Json)
    this.setBody(body)
}
```

**Comment by Bwvolleyball** [comment]
I can second this that this happened to me when using the `LinkTokenApi()`.
I got around this by customizing the ktor http client config with

```kotlin
                defaultRequest {
                    contentType(ContentType.Application.Json)
                }
```

But this feels like not a great way to handle this.

I agree with the above recommended implementation.

**Comment by Bwvolleyball** [comment]
Looks like this was resolved with `v2.0.5`: https://github.com/merge-api/merge-sdk-kotlin/pull/9/files#diff-aa2a04a7063510c81c0febe22a6d4259118a662012faaa8608200c0037fd1f16
